lib/kino_slack/message_cell.ex

```diff
@@ -0,0 +1,86 @@
+defmodule KinoSlack.MessageCell do
+  @moduledoc false
+
+  use Kino.JS, assets_path: "lib/assets"
+  use Kino.JS.Live
+  use Kino.SmartCell, name: "Slack"
+
+  @impl true
+  def init(attrs, ctx) do
+    fields = %{
+      "token_secret_name" => attrs["fields"]["token_secret_name"] || "",
+      "channel" => attrs["fields"]["channel"] || "",
+      "message" => attrs["fields"]["message"] || ""
+    }
+
+    ctx = assign(ctx, fields: fields)
+    {:ok, ctx}
+  end
+
+  @impl true
+  def handle_connect(ctx) do
+    {:ok, %{fields: ctx.assigns.fields}, ctx}
+  end
+
+  @impl true
+  def handle_event("update_channel", value, ctx) do
+    ctx = update(ctx, :fields, &Map.merge(&1, %{"channel" => value}))
+    {:noreply, ctx}
+  end
+
+  @impl true
+  def handle_event("update_message", value, ctx) do
+    ctx = update(ctx, :fields, &Map.merge(&1, %{"message" => value}))
+    {:noreply, ctx}
+  end
+
+  @impl true
+  def handle_event("update_token_secret_name", value, ctx) do
+    broadcast_event(ctx, "update_token_secret_name", value)
+    ctx = update(ctx, :fields, &Map.merge(&1, %{"token_secret_name" => value}))
+    {:noreply, ctx}
+  end
+
+  @impl true
+  def to_attrs(ctx) do
+    %{
+      "fields" => ctx.assigns.fields
+    }
```

Comment:
We generally do this:

```suggestion
    ctx.assigns.fields
```

Reply:
We group all editable attributes under fields just to make it easier to work with on the client, but `attrs` is what gets persisted in the notebook source, and it should be agnostic to how we end up with those attributes :) On a related note, `attrs` is what we always need to keep in mind when changing smart cells, because changes may impact compatibility.